sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIO.java

```diff
@@ -0,0 +1,1107 @@
+      return new AutoValue_FhirIO_Write.Builder().setFhirStore(fhirStore);
+    }
+
+    /**
+     * Create Method creates a single FHIR resource. @see <a
+     * href=https://cloud.google.com/healthcare/docs/reference/rest/v1beta1/projects.locations.datasets.fhirStores.fhir/create></a>
+     *
+     * @param fhirStore the hl 7 v 2 store
+     * @param gcsTempPath the gcs temp path
+     * @param gcsDeadLetterPath the gcs dead letter path
+     * @param contentStructure the content structure
+     * @return the write
+     */
+    public static Write fhirStoresImport(
+        String fhirStore,
+        String gcsTempPath,
+        String gcsDeadLetterPath,
+        @Nullable FhirIO.Import.ContentStructure contentStructure) {
+      return new AutoValue_FhirIO_Write.Builder()
+          .setFhirStore(fhirStore)
+          .setWriteMethod(Write.WriteMethod.IMPORT)
+          .setContentStructure(contentStructure)
+          .setImportGcsTempPath(gcsTempPath)
+          .setImportGcsDeadLetterPath(gcsDeadLetterPath)
+          .build();
+    }
+
+    public static Write fhirStoresImport(
+        ValueProvider<String> fhirStore,
+        ValueProvider<String> gcsTempPath,
+        ValueProvider<String> gcsDeadLetterPath,
+        @Nullable FhirIO.Import.ContentStructure contentStructure) {
+      return new AutoValue_FhirIO_Write.Builder()
+          .setFhirStore(fhirStore.get())
+          .setWriteMethod(Write.WriteMethod.IMPORT)
+          .setContentStructure(contentStructure)
+          .setImportGcsTempPath(gcsTempPath.get())
+          .setImportGcsDeadLetterPath(gcsDeadLetterPath.get())
+          .build();
+    }
+
+    /**
+     * Execute Bundle Method executes a batch of requests as a single transaction @see <a
+     * href=https://cloud.google.com/healthcare/docs/reference/rest/v1beta1/projects.locations.datasets.fhirStores.fhir/executeBundle></a>.
+     *
+     * @param fhirStore the hl 7 v 2 store
+     * @return the write
+     */
+    public static Write executeBundles(String fhirStore) {
+      return new AutoValue_FhirIO_Write.Builder()
+          .setFhirStore(fhirStore)
+          .setWriteMethod(WriteMethod.EXECUTE_BUNDLE)
+          .build();
+    }
+
+    /**
+     * Execute bundles write.
+     *
+     * @param fhirStore the fhir store
+     * @return the write
+     */
+    public static Write executeBundles(ValueProvider<String> fhirStore) {
+      return new AutoValue_FhirIO_Write.Builder()
+          .setFhirStore(fhirStore.get())
+          .setWriteMethod(WriteMethod.EXECUTE_BUNDLE)
+          .build();
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(Write.class);
+
+    @Override
+    public Result expand(PCollection<String> input) {
+      PCollection<HealthcareIOError<String>> failedBundles;
+      PCollection<HealthcareIOError<String>> failedImports;
+      switch (this.getWriteMethod()) {
+        case IMPORT:
+          LOG.info(
```

Comment:
A warning may work here. Up to you, though. sorry about the back-and-forth.  : )